design-proposals/multiarch-cluster.md

```diff
@@ -0,0 +1,154 @@
+
+**Virt-launcher**
+
+- Given that the virt-launcher pod runs on a per VM basis, relying on runtime.GOARCH is an acceptable
+
+**Virt-handler**
+
+- Given that the virt-handler runs as a daemonset, relying on runtime.GOARCH is acceptable
+
+
+## API Examples
+
+**Example VM Spec**
+
+```yaml
+apiVersion: kubevirt.io/v1alpha3
+kind: VirtualMachine
+metadata:
+  name: mytestvm
+spec:
+  running: true
+  template:
+    spec:
+      architecture: arm64
+      domain:
+        devices:
+          disks:
+          - name: disk1
+            disk:
+              bus: virtio
+          interfaces:
+          - name: network1
+            bridge: {}
+      networks:
+              - name: network1
+                multus:
+                  networkName: mytestvm/mytestvm
+      volumes:
+           - name: disk1
+             emptyDisk:
+               capacity: 50G
+```
+
+**Example Kubevirt Configuration**
+
+```yaml	
+apiVersion: kubevirt.io/v1
+kind: KubeVirt
+metadata:
+  name: kubevirt
+  namespace: kubevirt
+spec:
+  certificateRotateStrategy: {}
+  configuration:
+    developerConfiguration:
+      featureGates: ["MultiArchitecture"]
+    archConfiguration:
+      arm64:
+        emulatedMachines: ["virt"]
+        ovmfPath: "/usr/share/AAVMF"
+      amd64:
+        emulatedMachines: ["pc-*"]
+        machineType: "q35"
+    defaultVMIArchitecture: "amd64"
+  customizeComponents: {}
+  imagePullPolicy: IfNotPresent
+```
+
+## Scalability
+
+The proposed changes have no anticipated impact on scalability capabilities of the KubeVirt framework
+
+
+## Update/Rollback Compatibility
+
+The only issue as far as rollback compatibility goes would be around the KubeVirt components within k8s no longer being compiled for multi-architecture environments. If a user were to downgrade their deployment of KubeVirt while still running a multi-architecture cluster, there may be deployment issues for the core components but this would be an expected outcome. 
+
+## Functional Testing Approach
+
+Currently, the only area of this proposal that still requires clarity is what direction the KubeVirt maintainers would like to proceed with regarding functional testing. The existing cluster that is stood up as part of the functional tests leverages a helper script from the k8s repo that checks system architecture prior to bringing up a cluster. How to leverage this script but bring up a multi-architecture cluster still needs discussion.
+
+# Implementation Phases
+
+Proposed Implementation Rollout involves 2 stages:
+
+1. Merge code changes. This will allow us to still test that pods are created in the e2e test for the expected architecture, despite not being able to start up.
+2. Followup PR involving the pushing of multi-arch images as part of the release process
```

Comment:
Update the https://github.com/kubevirt/user-guide should be 2).  Image builds can be bumped to 3).